### ISSUE TITLE
ClaRa Type missmatch in function argument

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#8464]
### Description

For the developer version of ClaRa the model `ClaRa.Components.Mills.HardCoalMills.Check.combineMillandFurnace` is failing during type checking in the frontend. Here is the [log file](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.Mills.HardCoalMills.Check.combineMillandFurnace.err).

Same issue for

- ClaRa.Components.Mills.PhysicalMills.Check.TestMillBox_1
- ClaRa.Components.Mills.PhysicalMills.Check.TestMillBox_1_measurementInput
- ClaRa.Components.Mills.PhysicalMills.Check.TestMillBox_2
- ClaRa.Components.Mills.PhysicalMills.Volumes.Check.TestAerosolVolume
- ClaRa.Components.Mills.PhysicalMills.Volumes.Check.TestDryer
- ClaRa.Components.Mills.PhysicalMills.Volumes.Check.TestFuelJoin
- ClaRa.Components.Mills.PhysicalMills.Volumes.Check.TestGrinder

Could be an issue in the library or in OpenModelica.

### Steps to Reproduce

Simulate `ClaRa.Components.Mills.HardCoalMills.Check.combineMillandFurnace` with omc:

```
[/mnt/ReferenceFiles/ExtraLibs/packaged/ClaRa/Basics/Media/FuelObject.mo:40:4-44:46:writable] Error: Type mismatch for positional argument 4 in ClaRa.Basics.Media.FuelFunctions.heatCapacity_pTxi(fuelType=coalFlowSource.fuelObject.fuelModel). The argument has type:
  ClaRa.Basics.Media.FuelTypes.BaseFuel
expected type:
  ClaRa.Basics.Media.FuelTypes.Fuel_refvalues_v1
```

### Version and OS

- OpenModelica Version: OMCompiler v1.19.0-dev.545+g5c10d4c725
- OS: Ubuntu 20.04.3 LTS
- Versions of used Modelica libraries: ClaRa_dev @ c57bace0acbe86a8ae2ccf3e427c60093c5ac95c

**Comment by perost** [comment]
The issue is that `BaseFuel` contains arrays with a fixed sizes, while `Fuel_refvalues_v1` extends from `BaseFuel` and changes some of those sizes. So basically:
```Modelica
record BaseR
  constant Integer n = 3;
  Real x[n];
end BaseR;

record DerivedR
  extends BaseR(n = 5);
end DerivedR;

function f
  input BaseR r;
end f;

model M
  DerivedR r;
algorithm
  f(r);
end M;
```
I'm not sure if that's legal, since the function is expecting a record with arrays of a certain size but is given a record with arrays of a different size. The [specification](https://specification.modelica.org/master/functions.html#positional-or-named-input-arguments-of-functions) just says "The type of each argument must agree with the type of the corresponding parameter", whatever that means.

**Comment by casella** [comment]
I also think model `M` is not legal. I opened a request for clarification in ticket [#3103](https://github.com/modelica/ModelicaSpecification/issues/3103) of the Modelica Specification, please follow the discussion there.

**Comment by perost** [comment]
This seems to have been fixed in the library a week ago (commit 8baa8a1b), and all the models mentioned now pass the frontend.

**Comment by AnHeuermann** [comment]
Very nice. This was committed yesterday, so we will see if the next nightly test will show any improvements.

**Comment by perost** [comment]
> Very nice. This was committed yesterday, so we will see if the next nightly test will show any improvements.

The [report for ClaRa](https://libraries.openmodelica.org/branches/master/ClaRa_dev/ClaRa_dev.html) already seem to include the fix, there aren't any models that fail in the frontend due to this error at least. Many of the models now time out in the backend instead though, which causes issues with the reporting.

**Comment by casella** [comment]
> This seems to have been fixed in the library a week ago (commit 8baa8a1b), and all the models mentioned now pass the frontend.

@perost, I guess we missed that during today's meeting with XRG/TLK, that's why I pinged the MLS ticket again. Too many issues flying around in all directions 😄.

In any case, I think it's good to have that issue clarified.
